src/main/kotlin/net/maizegenetics/phgv2/cli/CreateRanges.kt

```diff
@@ -53,68 +65,79 @@ class CreateRanges: CliktCommand(help="Create BED file of reference ranges from
-            val modifiedFirst = if (first - pad < 0) 1 else first - pad
-            Pair((modifiedFirst - 1), (second + pad) )//Subtracting one from the start to do the conversion from GFF 1-based inclusive inclusive to BED's 0-based inclusive exclusive
+        }.map { (first, second,third) ->
+            // GFF is 1-based.  The BED file is 0-based, so we need to subtract 1 from the start position
+            Triple(Position(first.contig,first.position-1), Position(second.contig,second.position),third)
+        }
+
+        // Resolve overlaps/embedded regions
+        val geneRange: RangeMap<Position, String> = TreeRangeMap.create()
+        for (bound in boundMinMax) {
+            val boundRange = Range.closed(bound.first, bound.second)
+            // addRange will merge overlaps, toss embedded regions,
+            // and concat gene names when ranges are merged.
+            addRange(geneRange,boundRange,bound.third)
         }
 
-        return(boundMinMax)
+        // Add flanking regions if specified by user
+       // var flankingRange:RangeMap<Position,String> = TreeRangeMap.create()
+        val flankingRange = if (pad > 0) createFlankingList(geneRange, pad, refSeq) else geneRange
+
+        return flankingRange
     }
 
 
     /**
-     * Function to generate a list of BedRecord objects based on the input GFF file features.
-     *
-     * @param bounds A list of type `Pair<Int, Int>`. Generated from [idMinMaxBounds]`.
-     * @param genes A list of type `Gene`
-     * @param featureId Identifier key for value to pull from ID field of GFF. Defaults to `"ID"`
-     *
-     * @return A list of type `BedRecord`
-     *
+     * Function to generate a list of BedRecords based on a RangeMap of Position.
+     * Because overlapping genes were merged, the score and strand values of the
+     * BedRecord can not be determined and are set to 0 and "." respectively.
+     * Also, because of this, the number of bed records does not necessarily match
+     *  the number of genes in the GFF file.  THere may be fewer if overlaps were merged.
      */
-    fun generateBedRecords(bounds: List<Pair<Int, Int>>,
-                        genes: List<Feature>,
-                        featureId: String = "ID"
-    ) : List<BedRecord> {
-        return genes.mapIndexed { index, gene ->
-            BedRecord(
-                gene.seqid,
-                bounds[index].first,
-                bounds[index].second,
-                gene.attribute(featureId).first(),
-                if (gene.score?.isNaN() != false) 0 else gene?.score?.toInt()?:0,
-                when(gene.strand) {
-                    Strand.PLUS -> "+"
-                    Strand.MINUS -> "-"
-                    else -> "."
-                }
-            )
+    fun generateBedRecords(ranges: RangeMap< Position,String>): List<BedRecord> {
+
+        val bedRecordList = mutableListOf<BedRecord>()
+        // Are these sorted?
+        ranges.asMapOfRanges().forEach{ range ->
```

Comment:
I think you need something like the following:
`ranges.asMapOfRanges().entries().toList().sortedBy{it.key.lowerEndpoint()}.map{range ->
    bedRecordList.add(...)
}`

Reply:
@zrm22 I googled after I wrote the "are these sorted?" comment and the mapOfRanges are sorted based on lower Endpoint.  I can add the above but it might be superfluous - let me know.

Reply:
Yeah looking at the javadoc you are correct.  Maybe add a unit test just to be sure?

Reply:
sure - let me finish the createFlankingList junits first and I'll come back to this.